packages/filler/src/services/ContractInteractionService.ts

```diff
@@ -346,33 +336,74 @@ export class ContractInteractionService {
 				stateOverride: overrides as any,
 			})
 
-			console.log(`Gas estimate for filling order ${order.id} on ${order.destChain} is ${gas}`)
-
 			// Cache the results
-			this.cacheService.setGasEstimate(order.id!, gas, postGasEstimate, relayerFeeUSD)
+			this.cacheService.setGasEstimate(order.id!, gas, postGasEstimate, postGasEstimateFeeToken)
 
-			return { fillGas: gas, postGas: postGasEstimate, relayerFeeUSD: relayerFeeUSD }
+			return { fillGas: gas, postGas: postGasEstimate, relayerFeeInFeeToken: postGasEstimateFeeToken }
 		} catch (error) {
 			console.error(`Error estimating gas:`, error)
 			// Return a conservative estimate if we can't calculate precisely
-			return { fillGas: 3000000n, postGas: 270000n, relayerFeeUSD: 10000000n }
+			return { fillGas: 3000000n, postGas: 270000n, relayerFeeInFeeToken: 10000000n }
 		}
 	}
 
 	/**
-	 * Gets the current Native token price in USD
+	 * Gets the current Native token price (including Decimals)
 	 */
-	async getNativeTokenPriceUsd(chain: string): Promise<bigint> {
-		const { asset } = this.configService.getWrappedNativeAssetWithDecimals(chain)
-		const ethPriceUsd = await fetchTokenUsdPriceOnchain(asset)
+	async getNativeTokenPrice(chain: string): Promise<bigint> {
+		let client = this.clientManager.getPublicClient(chain)
+		const nativeToken = client.chain?.nativeCurrency
 
-		return ethPriceUsd
+		if (!nativeToken?.symbol || !nativeToken?.decimals) {
+			throw new Error("Chain native currency information not available")
+		}
+
+		const nativeTokenPriceUsd = await fetchTokenUsdPrice(nativeToken.symbol)
+
+		return BigInt(Math.floor(nativeTokenPriceUsd * Math.pow(10, 18)))
+	}
+
+	async convertGasToFeeToken(gasEstimate: bigint, chain: string): Promise<bigint> {
+		const client = this.clientManager.getPublicClient(chain)
+		const gasPrice = await client.getGasPrice()
+		const gasCostInWei = gasEstimate * gasPrice
+		const nativeToken = client.chain?.nativeCurrency
+
+		if (!nativeToken?.symbol || !nativeToken?.decimals) {
+			throw new Error("Chain native currency information not available")
+		}
+
+		const gasCostInToken = Number(gasCostInWei) / Math.pow(10, nativeToken.decimals)
+		const tokenPriceUsd = await fetchTokenUsdPrice(nativeToken.symbol)
+		const gasCostUsd = gasCostInToken * tokenPriceUsd
+
+		const { address: feeTokenAddress, decimals: feeTokenDecimals } = await this.getFeeTokenWithDecimals(chain)
+		const feeTokenPriceUsd = await fetchTokenUsdPrice("DAI") // Using DAI as default
```

Comment:
fee token is a stablecoin, so you should not have to fetch the price, just return gasCostUsd in the decimals of the fee token.

Reply:
Since this function just for the fee estimation, it's okay to assume a 1:1 peg